### PR TITLE
coreos-base/coreos-metadata: add initial ebuild

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by crawford** [comment]
The other approach I've been working on is to vendor the dependencies into the coreos-metadata repo. So far, I've only gotten it to work by cloning the index, but it should be possible to drop that restriction.

**Comment by crawford** [comment]
I was able to make this much simpler by vendoring all of the dependencies. Yes, it does make the repo more complex, but the upside is that we will never have a crate version mismatch.

**Comment by crawford** [comment]
Depends on https://github.com/coreos/portage-stable/pull/254.

**Comment by crawford** [comment]
Depends on https://github.com/coreos/coreos-overlay/pull/1441.

File: coreos-base/coreos-metadata/coreos-metadata-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+CROS_WORKON_PROJECT="coreos/coreos-metadata"
+CROS_WORKON_LOCALNAME="coreos-metadata"
+CROS_WORKON_REPO="git://github.com"
+inherit cros-workon systemd
+
+if [[ "${PV}" == 9999 ]]; then
+	KEYWORDS="~amd64 ~arm64"
+else
+	CROS_WORKON_COMMIT=""
+	KEYWORDS="amd64 arm64"
```
Comment (marineam) [inline coreos-base/coreos-metadata/coreos-metadata-9999.ebuild]: Have you verified that cross-compiling arm64 works correctly?

Comment (marineam) [inline coreos-base/coreos-metadata/coreos-metadata-9999.ebuild]: If not don't include arm64 in the keywords

File: coreos-base/coreos-metadata/files/coreos-metadata.service [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=CoreOS Metadata Agent
+
+[Service]
+Type=oneshot
+EnvironmentFile=-/usr/share/oem/oem-release
```
Comment (marineam) [inline coreos-base/coreos-metadata/files/coreos-metadata.service]: Why make oem-release file optional? Won't the command fail if OEM_ID isn't set?

File: coreos-base/coreos-metadata/coreos-metadata-9999.ebuild (hunk 1 above)
Comment (crawford) [inline coreos-base/coreos-metadata/coreos-metadata-9999.ebuild]: Removed

File: coreos-base/coreos-metadata/files/coreos-metadata.service (hunk 2 above)
Comment (crawford) [inline coreos-base/coreos-metadata/files/coreos-metadata.service]: Required

**Comment by marineam** [comment]
LGTM

**Pull request merged**